handlers_global.go

```diff
@@ -0,0 +1,76 @@
+package veneur
+
+import (
+	"compress/zlib"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+type contextHandler func(c context.Context, w http.ResponseWriter, r *http.Request)
+
+func (ch contextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// nil context is bad form, but since we don't use it, it's a quick hack
+	// to allow us to write tests before we upgrade to Go 1.7 Context type
+	// and the newer goji
+	// TODO(aditya) actually update this
+	ch(nil, w, r)
+}
+
+// handleImport generates the handler that responds to POST requests submitting
+// metrics to the global veneur instance.
+func handleImport(s *Server) http.Handler {
```

Comment:
We could do it that way, but it makes more sense to have this as a non-method function, because this is the function we're actually testing in our tests. (Ideally, the function that's provided to our tests would be entirely independent of a running server - we can't do that here, because the handler itself requires the server's state as part of its response, but architecturally, we want to mimic that model, even if that means essentially prebinding the function). Eventually, we should be moving to a model in which a handler function is not bound to a server's state at all, but I'm punting on that until we get better test coverage first.

In other words, the only reason this function exists is to give us an unbound handler function that we can use in our tests, because our tests should (ideally) not be operating on bound methods.